### DISCUSSION TITLE
very high cpu usage suddenly from jinx--mod-check

### DISCUSSION THREAD

**Discussion by emacsomancer** [minad/jinx#180] [Q&A] [answered]
I'm not sure what's happened, as I've used jinx for quite a while without issue, and the last update seemed to have been a week ago, so I'm guessing it's an interaction with something, but I'm suddenly seeing Emacs-freezing (for 5-10 seconds) CPU issues related to Jinx:

````
     61943  90% - timer-event-handler
       61941  90%  - apply
       61599  90%   - jinx--timer-handler
       61598  90%    - jinx--check-pending
       61582  90%     - jinx--check-region
       61562  90%      - jinx--word-valid-p
       61552  90%         jinx--mod-check
           2   0%        jinx--regexp-ignored-p
          16   0%       jinx--find-visible-pending  
````

I found https://github.com/minad/jinx/issues/35 which seems very similar, but I see from that issue that the predicate in question was removed already. 

It's happening in org-mode, and seems perhaps linked to opening up :PROPERTIES: trays.

**Comment by minad** [comment]
Hi!

Can you please give more details? Which versions of Emacs, Org, Jinx and Enchant do you use exactly? Issue #35 is not related - it was due to the slow flyspell predicate, which we've stopped using here since. The odd thing in your report is that jinx--mod-check itself is expensive. jinx--mod-check is the native call to Enchant and is usually fast.

**Reply by emacsomancer** [reply]
Emacs-master (as of Sun 02 Jun 2024 12:20:44 AM CDT), Org 9.7.2, jinx-20240606.909, Enchant 2.8.0-1.

It really seems to be a problem with :PROPERTIES: trays. It'll take 5-10 seconds to open, and then still be very slow for another 5-10 seconds. 

If I revert to my old flyspell config and turn off jinx, the issue disappears (which is probably unsurprising, given that it seems to be about calls to Enchant). 

Here's what the one property tray looks like:

```
:PROPERTIES:
:ID:       a74d5938-5747-4e2a-871a-8d6019395617
:AUTHOR:   "emacsomancer"
:MAIL:     me@me.com
:ROAM_ALIASES: EMACS emacs "GNU Emacs"
:CREATION_TIME: 1630461465
:LAT-LONG: 0,0
:LOCATION: "nowhere"
:LOCATION_SRC: hardcoded-ethernet
:END:
#+title: Emacs
#+date: [2021-12-15 Wed 16:34]
#+source: [https://en.wikipedia.org/wiki/Emacs https://git.savannah.gnu.org/cgit/emacs.git/log/ https://en.wikipedia.org/wiki/GNU_Emacs]
````

**Comment by emacsomancer** [comment]
I think I might have figured it out - possibly a conflict with `org-appear`.

Edit: No, that doesn't seem to solve it, upon further testing.

**Comment by emacsomancer** [comment]
I think it's something about  Enchant 2.8.0. I have the same Emacs set up on several machines, and on the two Arch machines, with 2.8.0, I have the same problem, but not on a different machine (same Emacs config) with 2.6.2.

**Reply by minad** [reply]
Thanks. Does the problem also occur with Enchant 2.7? Can you please report this upstream on the Enchant repository? If there is a significant slow down in Enchant itself it can only be fixed there.

**Reply by emacsomancer** [reply, marked as answer]
I'll report upstream. [edit: filed at https://github.com/AbiWord/enchant/issues/386 ]

I checked Arch history: they updated to 2.7 in May and there was no issue until they updated to 2.8 a week ago.

**Comment by gety9** [comment]
@emacsomancer were you able to solve the issue? my org mode file is unusable when i call "consult-outline" - it freezes, but if i remove jinx, and replace it with flyspell - everything is perfect.

**Reply by emacsomancer** [reply]
what version of enchant are you on? upstream enchant fixed the issue after I reported with v2.8.1.

**Reply by gety9** [reply]
> what version of enchant are you on? upstream enchant fixed the issue after I reported with v2.8.1.

It says 2.8.2

```
enchant-2 -v
@(#) International Ispell Version 3.1.20 (but really Enchant 2.8.2)
```

**Reply by emacsomancer** [reply]
That's the same version I'm on, on a couple of different machines/distros and I don't seem to have freezes since v2.8.1. 

I wonder if it's an issue with consult.

I tried running `consult-outline` on a largeish Org file and I see what you mean in terms of everything becoming jittery and slow. I checked with the profiler turned on and see:

````
        8592  56% - command-execute
        8592  56%  - call-interactively
        4626  30%   - byte-code
        4573  30%    - read-extended-command
        4573  30%     - read-extended-command-1
        4573  30%      - completing-read
        4573  30%       - apply
        4573  30%        - completing-read@llama
        4573  30%         - #<primitive-function completing-read>
        4573  30%          - completing-read-default
        4573  30%           - apply
        4573  30%            - vertico--advice
        4573  30%             - apply
        4573  30%              - #<byte-code-function F8D>
        4573  30%               - read-from-minibuffer
        3148  20%                - timer-event-handler
        3148  20%                 - apply
        3089  20%                  - jinx--timer-handler
        3089  20%                   - jinx--check-pending
        3089  20%                      jinx--find-visible-pending
````

So there is indeed possibly some jinx involvement. (I suspect this is a separate issue though from the one I originally had.)
